### PR TITLE
Add consumption data for MX

### PR DESCRIPTION
**Pull request by unitrium**
## Issue

The consumption data for Mexico is available as a live data set. So we should start collecting data as soon as possible so that we can use it for future estimation models.

## Description

Adds a fetch consumption function in the mexico parser.
Split Baja California in Baja California and Baja California Sur as those two regions are not interconnected.

### Preview

Testing to be added

- [x]  Adding testing

- [x]  Redo geometries

### CODE REVIEW

File: parsers/MX.py [hunk 1]
```diff
@@ -228,6 +264,44 @@ def fetch_exchange(
...
+    logger: Logger = getLogger(__name__),
+) -> list:
+    """Gets the consumption data for a region using the live dashboard."""
+    # TODO the calls could be improved since we can get all the data in one call.
+    if session is None:
+        session = Session()
+    if target_datetime is not None:
+        raise NotImplementedError("This parser is not yet able to parse past dates")
+    response = session.get(MX_EXCHANGE_URL)
+    if response.status_code != 200:
```
Comment (VIKTORVAV99) [inline parsers/MX.py:281]: should we use response.ok here instead? 
```suggestion
    if not response.ok:
```

File: parsers/MX.py [hunk 2]
```diff
@@ -228,6 +264,44 @@ def fetch_exchange(
...
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    """Gets the consumption data for a region using the live dashboard."""
+    # TODO the calls could be improved since we can get all the data in one call.
+    if session is None:
+        session = Session()
+    if target_datetime is not None:
+        raise NotImplementedError("This parser is not yet able to parse past dates")
+    response = session.get(MX_EXCHANGE_URL)
```
Comment (VIKTORVAV99) [inline parsers/MX.py:280]:
```suggestion
    response: Response = session.get(MX_EXCHANGE_URL)
```

**Review by VIKTORVAV99 — COMMENTED** [review]
Some initial comments and suggestions.

Just a note as well, I can't seem to access the consumption page locally. So please do double check that the parser server can access the page (it should be able to) otherwise we will have to set up a proxy first.

**Comment by unitrium** [comment]
I have updated the map, thanks!
Regarding the access to the page, it shouldn't be an issue since the page is the same from which we get the exchanges at the moment.

File: config/zones/MX-CE.yaml [hunk 3]
```diff
@@ -4,3 +4,5 @@ bounding_box:
   - - -96.22812130395499
     - 21.325344143417453
 timezone: null
```
Comment (VIKTORVAV99) [inline config/zones/MX-CE.yaml:6]: Should we add the timezone here?
```suggestion
timezone: America/Tijuana
```

File: config/zones/MX-NE.yaml [hunk 4]
```diff
@@ -4,3 +4,5 @@ bounding_box:
   - - -96.63927161399994
     - 28.293549303557484
 timezone: null
```
Comment (VIKTORVAV99) [inline config/zones/MX-NE.yaml:6]:
```suggestion
timezone: America/Tijuana
```

File: config/zones/MX-NO.yaml [hunk 5]
```diff
@@ -4,3 +4,5 @@ bounding_box:
   - - -99.30696277711047
     - 32.277751364000025
 timezone: null
```
Comment (VIKTORVAV99) [inline config/zones/MX-NO.yaml:6]:
```suggestion
timezone: America/Tijuana
```

File: config/zones/MX-NW.yaml [hunk 6]
```diff
@@ -4,3 +4,5 @@ bounding_box:
   - - -104.88862626838755
     - 32.99936330200008
 timezone: null
```
Comment (VIKTORVAV99) [inline config/zones/MX-NW.yaml:6]:
```suggestion
timezone: America/Tijuana
```

File: config/zones/MX-OC.yaml [hunk 7]
```diff
@@ -4,3 +4,5 @@ bounding_box:
   - - -97.5001883613038
     - 25.66554433840048
 timezone: null
```
Comment (VIKTORVAV99) [inline config/zones/MX-OC.yaml:6]:
```suggestion
timezone: America/Tijuana
```

File: config/zones/MX-OR.yaml [hunk 8]
```diff
@@ -4,3 +4,5 @@ bounding_box:
   - - -89.87985327199993
     - 22.98054026911808
 timezone: null
```
Comment (VIKTORVAV99) [inline config/zones/MX-OR.yaml:6]:
```suggestion
timezone: America/Tijuana
```

File: config/zones/MX-PN.yaml [hunk 9]
```diff
@@ -4,3 +4,5 @@ bounding_box:
   - - -86.20059160099993
     - 22.123439846000053
 timezone: null
```
Comment (VIKTORVAV99) [inline config/zones/MX-PN.yaml:6]:
```suggestion
timezone: America/Tijuana
```

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

Just have some suggestions to add timezones but I think everything looks good code wise. Note: I did not run this locally as my access to the source is blocked.

**Pull request merged**